logging/logger.go

```diff
@@ -51,6 +53,28 @@ var (
 	logRotateBytes    int
 )
 
+// To check if the provided log file path is valid or not
+func validateLogFilePath(logFileDir string, ui cli.Ui) bool {
+	_, err := os.Stat(logFileDir)
+
+	if err != nil {
+		ui.Error(fmt.Sprintf("Cannot access directory %v", err))
+		return false
+	}
+
+	var stat syscall.Stat_t
+	err = syscall.Stat(logFileDir, &stat);
+
+	if err == nil {
+		if uint32(os.Geteuid()) != stat.Uid {
```

Comment:
This does not mean logs cannot be written: write permission could be granted by group membership for instance: if `/var/logs/consul/` has permissions `drwxrwxr-x` and is owned by `root` and group `consul` (assuming `consul` user is member of `consul` group)

Furthermore, this is not compatible with Windows permissions system

Reply:
> This does not mean logs cannot be written: write permission could be granted by group membership for instance: if `/var/logs/consul/` has permissions `drwxrwxr-x` and is owned by `root` and group `consul` (assuming `consul` user is member of `consul` group)
> 
> Furthermore, this is not compatible with Windows permissions system

my bad, I will remove the permission check part as of now and would try to think of a better approach. If you already have something in your mind, let me know I would love to work on that too.
Thanks for pointing this out.

Reply:
@djmgit AFAIK, this is very hard, because there are not only permissions for user/group/other, you have to take into account Posix ACL, optionally partition being read-only... so just on POSIX systems, this is very hard (not even talking about Windows)
Personally, I would simply try to create a file and it if fails, report the error (and do not forget to delete the file)

Reply:
> @djmgit AFAIK, this is very hard, because there are not only permissions for user/group/other, you have to take into account Posix ACL, optionally partition being read-only... so just on POSIX systems, this is very hard (not even talking about Windows)
True
> Personally, I would simply try to create a file and it if fails, report the error (and do not forget to delete the file)
Yeah, I too thought of this, though it will have the overhead of creating and deleting a dummy file, however this method will cover both the cases, both non existent path and permission issue.